gateway/ln-gateway/src/client.rs

```diff
@@ -22,3 +27,56 @@ dyn_newtype_define! {
   #[derive(Clone)]
   pub GatewayClientBuilder(Arc<IGatewayClientBuilder>)
 }
+
+#[derive(Debug, Clone)]
+pub struct RocksDbGatewayClientBuilder {
+    pub work_dir: PathBuf,
+}
+
+/// Default gateway clinet builder which constructs clients with RocksDb
+/// and saves the config at the given work directory
+impl RocksDbGatewayClientBuilder {
+    pub fn new(work_dir: PathBuf) -> Self {
+        Self { work_dir }
+    }
+}
+
+/// Builds a new federation client with RocksDb
+/// On successful build, the configuration is saved to a file at the builder work directory
+impl IGatewayClientBuilder for RocksDbGatewayClientBuilder {
+    fn build(&self, config: GatewayClientConfig) -> Result<Client<GatewayClientConfig>> {
+        let federation_id = FederationId(config.client_config.federation_name.clone());
+
+        let db = self.create_database(federation_id)?;
+        let ctx = secp256k1::Secp256k1::new();
+
+        Ok(Client::new(config, db, ctx))
+    }
+
+    /// Create a client database
+    fn create_database(&self, federation_id: FederationId) -> Result<Database> {
+        let db_path = self.work_dir.join(format!("{}.db", federation_id.0));
```

Comment:
Since the federation ID can be set by the federation this can be used to launch a [path traversal attack](https://owasp.org/www-community/attacks/Path_Traversal) against gateways.

Please lmk if you want to fix it in this PR or create a high priority issue. Possible fixes:

* Make the id deterministically derived from the guardian's pub keys as discussed elsewheere
* Hash the federation name
* Sanitize the federation name (probably not effective in all circumstances, I'd avoid this)

Reply:
Ack.

My preference is to hash the value of FederationID as an immediate fix, then depending on how we implement #754 , reassess whether we should keep the hashing step or to remove it.

I'll introduce the hash step as part of this PR